docs/strategies/gradle.md

```diff
@@ -1,29 +1,190 @@
+
+#### Relationship to analysis targets
+
+Each pair of `(subproject, configuration)` corresponds to one dependency graph.
+
+Ideally, each `(subproject, configuration)` pair would be a separate analysis target. For technical reasons, the current implementation treats separate subprojects as analysis targets, and treats each subproject as the union of all of its configurations.
+
+In practice, this should not affect returned dependency results.
+
+### Gradle wrappers
+
+Instead of invoking `gradle` directly, most Gradle projects use a "Gradle wrapper", which is a shell script vendored in the project that selects and downloads the correct version of Gradle. See [The Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) for details.
+
+Because of this, the Gradle analyzer has logic for selecting which `gradle` executable to actually use. See [Running Gradle](#running-gradle) for details.
+
+## Running Gradle
+
+This strategy requires dynamic analysis in its discovery phase (not just in the analysis phase). This is because we need to execute Gradle in order to list subprojects and evaluate `build.gradle` files.
+
+When executing Gradle for an analysis target at directory `ANALYSIS_TARGET_DIR`, the CLI will prefer (in order):
+
+1. `$ANALYSIS_TARGET_DIR/gradlew`
+2. `$ANALYSIS_TARGET_DIR/gradlew.bat`
+3. `gradle` (from `$PATH`)
+
+For more details, see [Gradle wrappers](#gradle-wrappers).
+
+In this documentation below, for brevity, we'll always refer to the selected tool as `gradle`.
+
+## Discovery
+
+This strategy discovers analysis targets by looking for files in the folder being analyzed whose names start with `build.gradle`. This matches both `build.gradle` as well as `build.gradle.kts` and build files in other Gradle-supported languages (`build.gradle.*`).
+
+It then executes `gradle projects` in the directory where the build file is found to get a list of subprojects for this Gradle build. These subprojects are used to create the analysis targets for this Gradle build.
+
+If there are no subprojects, a analysis target is created that analyzes the root project. Otherwise, a set of analysis targets is created: one for each Gradle subproject.
+
+## Tactics
+
+### Tactic selection
+
+This strategy selects tactics by trying them in preference order and using the results of the first tactic that succeeds.
+
+The order of tactics for this strategy is:
+
+1. Gradle build plugin
+2. Parsing `gradle :dependencies` (not yet implemented)
+
+### Gradle build plugin
+
+|                    |                                                                                          |
+| ------------------ | ---------------------------------------------------------------------------------------- |
+| :heavy_check_mark: | This tactic reports dependencies for all subprojects.                                    |
+| :heavy_check_mark: | This tactic provides a graph for subproject dependencies.                                |
```

Comment:
I'm not sure I would understand the importance of this as an end-user

Reply:
I think power users will understand this. I'm also open to a better explanation, but I can't think of one right now.